cmd/lifecycle/analyzer.go

```diff
@@ -204,7 +205,7 @@ func (a *analyzeCmd) Exec() error {
 		return err
 	}
 
-	if err := lifecycle.WriteTOML(a.analyzedPath, analyzedMD); err != nil {
+	if err := buildpack.WriteTOML(a.analyzedPath, analyzedMD); err != nil {
```

Comment:
Why did this function move to the buildpack package? I don't think we want the `main` package to know about buildpacks. It should only know about the platform...

Reply:
The same `WriteTOML` function existed in both the `lifecycle` and `buildpack` packages. Since we import the `buildpack` package anyway in most of the files where we used to call `WriteTOML` under the `lifecycle` package, I thought it might be useful to just use the `buildpack` package.
As you can see, I've made this change in a separate commit so it'll be easier to make changes to it if needed.

Reply:
I still think it's weird that `main` should depend on the `buildpack` package. `WriteTOML` is a very simple function, maybe we could just duplicate it.

Reply:
I understand the weirdness but I really don't like duplicating code in such case.
I found this issue because the new package that I created is also using this function as well and in order to prevent circular imports, we can either use the buildpack implementation or duplicate it again (for the third time).
What do you think about creating a helper package that all three packages will be able to use?

Reply:
I think it's okay for `layermetadata` to use `buildpack.WriteTOML` (because it needs to import `buildpack` anyway). But I don't think we need to remove `lifecycle.WriteTOML`. That is two copies of the function, which seems okay. Then `main` can use `lifecycle.WriteTOML`, and avoid importing `buildpack`.

Reply:
As you can see in [this commit](https://github.com/buildpacks/lifecycle/pull/724/commits/b08737e7b89539676e9ad709915d643e22d109b6), all of the `main`, except the `rebaser` already import `buildpack` (even without calling the `WriteTOML` function). This is why I thought there is no problem with using the `buildpack` package also for this purpose.

Reply:
You are right! I guess it just reads weird, since `buildpack.WriteTOML` implies that a buildpack is doing something, whereas this action has nothing to do with buildpacks.

Reply:
Please see https://github.com/buildpacks/lifecycle/pull/724/commits/ced3b85ee7ae01a12682461f0fc549a70cb09c48.
What do you think about putting it in a new package?

Reply:
I'm going to mark this conversation as resolved. Let's continue the discussion [here](https://github.com/buildpacks/lifecycle/pull/724#discussion_r717648603).